src/vfs.rs

```diff
@@ -820,6 +820,28 @@ impl DavFile for AliyunDavFile {
         .boxed()
     }
 
+    fn redirect_url(&mut self) -> FsFuture<Option<String>> {
+        debug!(file_id = %self.file.id, file_name = %self.file.name, "file: redirect_url");
+        async move {
+            if self.file.id.is_empty() {
+                return Err(FsError::NotFound);
+            }
+            let download_url = self.file.url.take();
+            let download_url = if let Some(mut url) = download_url {
+                if is_url_expired(&url) {
+                    debug!(url = %url, "download url expired");
+                    url = self.get_download_url().await?.url;
+                }
+                url
+            } else {
+                let res = self.get_download_url().await?;
+                res.url
+            };
+            Ok(Some(download_url))
```

Comment:
`.livp` 的情况下 download_url 可能为空。

```suggestion
            if !download_url.is_empty() {
                Ok(Some(download_url))
            } else {
                Ok(None)
            }
```